ui/src/components/vault/VaultManagement/VaultManagement.jsx

```diff
@@ -52,38 +53,38 @@ const VaultManagement = () => {
 
   const { state, walletP } = useApplicationContext();
 
-  const {
-    purses,
-    vaults,
-    vaultToManageId,
-    brandToInfo,
-    treasury: { priceAuthority },
-  } = state;
+  const { purses, vaults, vaultToManageId, brandToInfo, treasury } = state;
 
   /** @type { VaultData } */
-  let vaultToManage = {
-    debt: null,
-    interestRate: null,
-    liquidationRatio: null,
-    locked: null,
-  };
+  let vaultToManage;
   if (vaultToManageId && vaults) {
     vaultToManage = vaults[vaultToManageId];
+  } else {
+    return <Redirect to="/vaults" />;
   }
+  const {
+    interestRate,
+    liquidationRatio,
+    locked,
+    debtSnapshot,
+  } = vaultToManage;
 
-  const { debt, interestRate, liquidationRatio, locked } = vaultToManage;
-
-  if (locked === null || debt === null) {
+  if (!locked || !debtSnapshot) {
     return <Redirect to="/vaults" />;
   }
-  assert(locked && debt);
+  assert(locked && debtSnapshot);
```

Comment:
pretty sure this is redundant too since if either of these were falsy that line 73 would leave this function

Reply:
So I needed to add back in the undefined for reasons stated in my other comment, making these asserts necessary, but I was at least able to remove the `null`s from the `VaultData` type now